main.js

```diff
@@ -63,13 +103,13 @@ function enableButton() {
 
 function toggleStar(event) {
   var id = event.target.parentNode.parentNode.parentNode.id;
-  console.log(event);
 
   for (var i = 0; i < allCards.length; i++) {
     if(allCards[i].id.toString() === id) {
       allCards[i].starred = !allCards[i].starred;
     }
   }
+  cardStorageRefresh();
```

Comment:
Could you clarify why we are completely clearing out local storage when trying to modify just one card? Can't we just pull that one card, modify it, then plug it back into local storage?

Reply:
Sure, my thought process was that in order to modify one card, we would have to iterate through local storage, then remove that card, then re-add that card (from what I could find there isn't a modify property, things are generally done with overwrites). 

Since allCards should always be in sync with storage as:
1. we load all cards from local storage into the allCards array on page load
2. any new cards created are added to allCards
3. any cards deleted are removed from allCards
4. any card changes that occur will be reflected in allCards

It seemed a more efficient approach to just clear everything out and re-add them all, and in my mind was slightly less error prone.

Reply:
Gotcha, I totally understand what you're doing there and I like that idea. Would it be possible and slightly easier if we cut out the allCards array and just made local storage an array that's our only source of storing cards? For example, the key we would use in local storage is "all_cards" and the value would then be the array of cards. We already have the functionality working to loop through arrays, so it would be a nice refactor if we just did that to the local storage.

Reply:
If I'm understanding you correctly, I think we might run into two problems with that approach:

1. This problem is actually a live one we just haven't run up against, which is that the cards we pull down from local storage are objects, but I think because of the way JSON works (it can't take functions), they lose their methods, we might need to reinstantiate them and aren't doing that now. In reinstancting them we'll need to store them in a local array.

2. Also as local storage is always stored in the string format, you'll always need to at least temporarily store that value to convert it over. You could maybe do this on the fly for each card on a specific interaction? But that might actually work out to be more steps than what we're doing now. Why don't we jump on a call and try run through it and see.